stream-chat-android-client/src/main/java/io/getstream/chat/android/client/StreamLifecycleObserver.kt

```diff
@@ -46,12 +46,17 @@ internal class StreamLifecycleObserver(
         }
     }
 
-    suspend fun dispose() {
+    suspend fun dispose(inThread: Boolean) {
         logger.d { "[dispose] no args" }
         if (isObserving) {
-            withContext(DispatcherProvider.Main) {
+            if (inThread) {
                 lifecycle.removeObserver(this@StreamLifecycleObserver)
                 logger.v { "[dispose] unsubscribed" }
+            } else {
+                withContext(DispatcherProvider.Main) {
+                    lifecycle.removeObserver(this@StreamLifecycleObserver)
+                    logger.v { "[dispose] unsubscribed" }
+                }
             }
         }
         isObserving = false
```

Comment:
Can we check if this method is invoked from the `Main` thread to avoid passing any additional flags?